Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,289 @@
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (array == nil) {
+        return [self init];
+    }
+
+    THROW_NS_IF_FALSE(E_BOUNDS, NSMaxRange(range) <= [array count]);
+
+    if (self = [self init]) {
+        NSUInteger start = range.location;
+        NSUInteger end = NSMaxRange(range);
+
+        if (flag) {
+            for (NSUInteger i = start; i < end; i++) {
+                id obj = [[array objectAtIndex:i] copy];
+                [self _insertObject:obj];
+                [obj release];
+            }
+        } else {
+            for (NSUInteger i = start; i < end; i++) {
+                [self _insertObject:[array objectAtIndex:i]];
+            }
+        }
+    }
+    return self;
+}
+
+- (void)_insertObject:(id)object {
+    if ([self containsObject:object]) {
+        return;
+    }
+
+    [_arrayContainer addObject:object];
+    [_setContainer addObject:object];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (NSUInteger)count {
+    return [_arrayContainer count];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSArray*)array {
+    return [[[_NSProxyOrderedSetArray alloc] initWithOrderedSet:self] autorelease];
+}
+
+/**
+ @Status Caveat
+ @Notes returned set does not reflect future changes.
+*/
+- (NSSet*)set {
+    return [[_setContainer copy] autorelease];
+}
+
+/**
+ @Status Interoperable
 */
 - (instancetype)initWithObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    THROW_NS_IF_FALSE(E_INVALIDARG, object != nil);
+    id ret[] = { object };
+    return [self initWithObjects:ret count:1];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObjects:(id)firstObj {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithObjects:(id)firstObj, ... {
+    va_list argList;
+    va_start(argList, firstObj);
+    std::vector<id> flatArgs = ConvertVAListToVector((id)firstObj, argList);
+    va_end(argList);
+    return [self initWithObjects:flatArgs.data() count:flatArgs.size()];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)cnt {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    THROW_NS_IF_FALSE(E_INVALIDARG, objects != nil);
```

Comment:
objects can be NULL if count is 0.

(Also prefer NULL or nullptr for non-ObjC pointers.)